lib/routes/51cto/recommend.ts

```diff
@@ -38,15 +80,28 @@ async function handler(ctx) {
         },
     });
     const list = response.data.data.data.list;
+
+    const browser = await puppeteer();
+    // let page = await init_page(browser);
+    // Create a list that includes 5 pages
+    const numPages = 50;
+    const pages = Array.from({ length: numPages }, () => init_page(browser));
+    let n = 0;
+
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.url, async () => {
+                n++;
+                return get_fullcontent(item, await pages[n % numPages]);
+            })
+        )
+    );
+
+    browser.close();
```

Comment:
Please parse the HTML using `got` and `cheerio`. I don't think puppeteer is necessary.

Reply:
I have tested that request without using `puppeteer` will keep returning things like the image down below:
![image](https://github.com/user-attachments/assets/71a557ce-1807-4b28-93bf-d5223aac30df)

Reply:
This is because the webpage expects you have finished the JS challenge and have the cookie `__tst_status` with proper value.

Reply:
I think it's impossible to figure out what's the meaning of these code.
These values change randomly:
![image](https://github.com/user-attachments/assets/f5410b88-18e7-457f-af4e-c8e822e923df)

Reply:
I have tried to make LLM to deobfuscate these code, but we still need to figure out how these values inside the code change.

<details>
  <summary>The result of deobfuscating</summary>
  <pre><code>
const strings = ['wyeCN', '4|2|0|3|1', 'wQzOV', 'split', 'iTyzs', 'pCQRM', 'cookie', 'location.href=location.href.replace(/[?|&]tads/', ' \'\')', 'WTKkN', 'bOYDu', 'dtzqS'];

function getString(index) {
    return strings[index];
}
export function deobfuscate() {
    const constants = {
        'WTKkN': 1086606777,
        'bOYDu': 346788022,
        'dtzqS'(a, b) { return a + b; },
        'wyeCN': 1370109662,
        'pCQRM'(a) { return a(); }
    };

    let result = 0;
    result += constants.WTKkN;
    result += constants.bOYDu;
    result = constants.dtzqS(result, constants.wyeCN);

    function process() {
        const operations = {
            'wQzOV': getString(4),
            'iTyzs'(a, b) { return a + b; }
        };

        const steps = operations.wQzOV.split('|');
        let i = 0;
        let value = 0;

        while (true) {
            switch (steps[i++]) {
                case '0': value += 217; break;
                case '1': return value;
                case '2': value += 164; break;
                case '3': value = operations.iTyzs(value, 448); break;
                case '4': value = 0; break;
            }
        }
    }

    constants.pCQRM(process);
    return result;
  </code></pre>
</details>

Reply:
You don't need to know how these values change, you can simply parse those 3 big numbers using regex and do what the JS challenge does, which is adding them up.

Reply:
I have tried my best to do this, but it doesn't work at all.

<details>
  <summary>The result</summary>
   
![code](https://github.com/user-attachments/assets/63b780bc-1e98-41a4-9e36-b19b13fce95f)

</details>

Reply:
> which is adding them up.

Sorry, I didn't see your message. Oh god, I can't imagine it just add two numbers together. 😂

Reply:
It's working now!